### DISCUSSION TITLE
How to construct a nested object back from javers audit for viewing the changes made.

### DISCUSSION THREAD

**Discussion by ketannabera** [javers/javers#1085] [Q&A]
Using javers-spring-boot-starter-mongo:5.11.2, with spring boot mongo application.
Let's say , we have a nested object :
Person{
       name; age; 
       cars :[
             {name; model;
              accessories :[accessory1,accessory2]
             },
             {name; model;
              accessories :[accessory1,accessory2]
             }
       ]
}

A person object which has a list of Car objects. Each car object in turn has a list of accessories objects. 
Now, all these objects are individually editable inside the nested structure and Javers perfectly audits this by linking with Value objects,fragments etc.

Now, the requirement is to build this Person object back from the audit so as to compare this object's entire audit history.
We are building a lot of complex code on the Java side by fetching snapshots,changes etc linked to this person. So is there a way to get this out of the box.
I am just wondering that if it can break this big object into fragments while auditing , then can it be reconstructed back in the javers code somehow ?

**Comment by bartoszwalacik** [comment]
Looks like you need Shadows
https://javers.org/documentation/jql-examples/#query-for-shadows
